mautic/mautic
# Prevent conditions connected into the "inaction/negative/red" path of a decision from being evaluated if the contact went down the "action/positive/green" path

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

When a condition is attached to a decision's red anchor (inactive) and a contact makes a decision that sends them down the green anchor (active) path, the condition is evaluated and logged even though any actions attached to the condition is not. This prevents that from happening. 


[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a campaign similar to this: ![](http://alan.direct//drop/2018-01-09_15-29-16.png)
2. Send a contact through the campaign 
3. Open the email sent to a contact in a visitor/anonymous browser or device
4. Look at the campaign and note that the Is:US condition was processed under the conditions tab

#### Steps to test this PR:
1. Repeat and the condition should not be processed. 
2. Run the new functional tests
